File: node_backend/modals/home/allRestaurants.js
import mongoose from "mongoose";

const { Schema } = mongoose;

const imageBadgeSchema = new Schema({
  imageId: String,
  description: String,
});

const textExtendedBadgeSchema = new Schema({
  iconId: String,
  shortDescription: String,
  fontColor: String,
});

const badgesSchema = new Schema({
  imageBadges: [imageBadgeSchema],
  textExtendedBadges: [textExtendedBadgeSchema],
});

const slaSchema = new Schema({
  deliveryTime: Number,
  lastMileTravel: Number,
  serviceability: String,
  slaString: String,
  lastMileTravelString: String,
  iconType: String,
});

const availabilitySchema = new Schema({
  startTime: String,
  nextCloseTime: String,
  opened: Boolean,
});

const aggregatedDiscountInfoV3Schema = new Schema({
  header: String,
  subHeader: String,
  discountTag: String,
});

const restaurantSchema = new Schema({
  aggregatedDiscountInfoV3: aggregatedDiscountInfoV3Schema,
  areaName: String,
  availability: availabilitySchema,
  avgRating: Number,
  avgRatingString: String,
  // badges: badgesSchema,
  badges: Boolean,
  cloudinaryImageId: String,
  costForTwo: String,
  cuisines: [String],
  isOpen: Boolean,
  name: String,
  sla: slaSchema,
  totalRatingsString: String,
  type: String,
  veg: Boolean,
  adminUserId: {
    type: mongoose.Types.ObjectId,
    ref: "AdminUser", // assuming 'AdminUser' is your admin user model name
    required: true,
  },
});

const AllRestaurants = mongoose.model("AllRestaurant", restaurantSchema);

export default AllRestaurants;

File: python_backend/app/models/home/all_restaurants.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from bson import ObjectId

class ImageBadge(BaseModel):
    imageId: str
    description: str

class TextExtendedBadge(BaseModel):
    iconId: str
    shortDescription: str
    fontColor: str

class Badges(BaseModel):
    imageBadges: List[ImageBadge]
    textExtendedBadges: List[TextExtendedBadge]

class Sla(BaseModel):
    deliveryTime: int
    lastMileTravel: float
    serviceability: str
    slaString: str
    lastMileTravelString: str
    iconType: str

class Availability(BaseModel):
    startTime: str
    nextCloseTime: str
    opened: bool

class AggregatedDiscountInfoV3(BaseModel):
    header: str
    subHeader: str
    discountTag: str

class Restaurant(BaseModel):
    aggregatedDiscountInfoV3: AggregatedDiscountInfoV3
    areaName: str
    availability: Availability
    avgRating: float
    avgRatingString: str
    badges: bool
    cloudinaryImageId: str
    costForTwo: str
    cuisines: List[str]
    isOpen: bool
    name: str
    sla: Sla
    totalRatingsString: str
    type: str
    veg: bool
    adminUserId: str
